### PR TITLE
refactor(fold): convert function comments to doxygen format

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/fold.c [hunk 1]
```diff
@@ -427,28 +408,22 @@ void opFoldRange(pos_T firstpos, pos_T lastpos, int opening, int recurse, int ha
...
 void openFold(pos_T pos, long count)
 {
   setFoldRepeat(pos, count, true);
 }
 
 // openFoldRecurse() {{{2
-/*
- * Open fold for current window at line "lnum" recursively.
- */
+/// Open fold for current window at line "lnum" recursively.
```
Comment (bfredl) [inline src/nvim/fold.c:419]: same

File: src/nvim/fold.c [hunk 2]
```diff
@@ -1181,10 +1133,8 @@ static void checkupdate(win_T *wp)
 }
 
 // setFoldRepeat() {{{2
-/*
- * Open or close fold for current window at line "lnum".
- * Repeat "count" times.
- */
+/// Open or close fold for current window at line "lnum".
```
Comment (bfredl) [inline src/nvim/fold.c:1136]: pos

File: src/nvim/fold.c [hunk 3]
```diff
@@ -1538,10 +1483,8 @@ static void foldMarkAdjustRecurse(win_T *wp, garray_T *gap, linenr_T line1, line
 }
 
 // getDeepestNesting() {{{2
-/*
- * Get the lowest 'foldlevel' value that makes the deepest nested fold in the
- * current window open.
- */
+/// Get the lowest 'foldlevel' value that makes the deepest nested fold in the
+/// current window open.
```
Comment (bfredl) [inline src/nvim/fold.c:1487]: `wp` and not current window.

File: src/nvim/fold.c [hunk 4]
```diff
@@ -1643,10 +1586,8 @@ static void setSmallMaybe(garray_T *gap)
 }
 
 // foldCreateMarkers() {{{2
-/*
- * Create a fold from line "start" to line "end" (inclusive) in the current
- * window by adding markers.
- */
+/// Create a fold from line "start" to line "end" (inclusive) in the current
+/// window by adding markers.
```
Comment (bfredl) [inline src/nvim/fold.c:1590]: same, not for current window

File: src/nvim/fold.c [hunk 5]
```diff
@@ -1672,9 +1613,7 @@ static void foldCreateMarkers(win_T *wp, pos_T start, pos_T end)
 }
 
 // foldAddMarker() {{{2
-/*
- * Add "marker[markerlen]" in 'commentstring' to line "lnum".
- */
+/// Add "marker[markerlen]" in 'commentstring' to line "lnum".
```
Comment (bfredl) [inline src/nvim/fold.c:1616]: pos

**Comment by bfredl** [comment]
otherwise LGTM

File: src/nvim/fold.c [hunk 6]
```diff
@@ -282,10 +276,11 @@ int foldLevel(linenr_T lnum)
   return foldLevelWin(curwin, lnum);
 }
 
-// lineFolded() {{{2
-// Low level function to check if a line is folded.  Doesn't use any caching.
-// Return true if line is folded.
-// Return false if line is not folded.
+/// lineFolded() {{{2
```
Comment (zeertzjq) [inline src/nvim/fold.c:279]: Comments like this do not seem like documentation comments.

Comment (dundargoc) [reply src/nvim/fold.c:279]: You're right. Thanks.

**Review by bfredl — APPROVED** [review]

File: src/nvim/fold.c [hunk 7]
```diff
@@ -378,9 +361,7 @@ void closeFold(pos_T pos, long count)
 }
 
 // closeFoldRecurse() {{{2
-/*
- * Close fold for current window at line "lnum" recursively.
- */
+/// Close fold for current window at line "lnum" recursively.
```
Comment (zeertzjq) [inline src/nvim/fold.c:364]: pos

File: src/nvim/fold.c [hunk 8]
```diff
@@ -427,28 +408,22 @@ void opFoldRange(pos_T firstpos, pos_T lastpos, int opening, int recurse, int ha
 }
 
 // openFold() {{{2
-/*
- * Open fold for current window at line "lnum".
- * Repeat "count" times.
- */
+/// Open fold for current window at line "lnum".
```
Comment (zeertzjq) [inline src/nvim/fold.c:411]: pos

**Pull request merged**